controllers/btpoperator_controller_test.go

```diff
@@ -111,21 +112,18 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
 	})
 
 	Describe("Provisioning", func() {
-		BeforeAll(func() {
+		BeforeEach(func() {
 			cr = createBtpOperator()
 			Eventually(k8sClient.Create(ctx, cr)).WithTimeout(k8sOpsTimeout).WithPolling(k8sOpsPollingInterval).Should(Succeed())
 		})
 
-		AfterAll(func() {
+		AfterEach(func() {
+			cr = &v1alpha1.BtpOperator{}
 			Eventually(k8sClient.Get(ctx, client.ObjectKey{Namespace: defaultNamespace, Name: btpOperatorName}, cr)).
 				WithTimeout(k8sOpsTimeout).
 				WithPolling(k8sOpsPollingInterval).
 				Should(Succeed())
 			Eventually(k8sClient.Delete(ctx, cr)).WithTimeout(k8sOpsTimeout).WithPolling(k8sOpsPollingInterval).Should(Succeed())
-			Eventually(getCurrentCrStatus).
-				WithTimeout(crStateChangeTimeout).
-				WithPolling(crStatePollingInterval).
-				Should(SatisfyAll(HaveField("State", types.StateDeleting), HaveField("Conditions", HaveLen(1))))
```

Comment:
this (and a few other state transition checks) would benefit from not using polling but instead validating on receival of events describing the state transition.

The reason is that checking a temporary state with a polling interval can have a sampling period larger than the transition itself - the first poll check could see the result of the unchanged state, the second poll check might see already the final state missing the transition state. This could be another factor contributing to higher flakiness.

For now removing the check for the transitional state and keeping only the check for the final state where `BTPOperator` custom resource is already gone. Will reintroduce the transitional checks back after redesigning to watch on events instead of polling.